File: 品味Java的21種設計模式/dp_cpp/templatemethod/example6/UserModel.cpp
#include "UserModel.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace templatemethod
			{
				namespace example6
				{

					std::string UserModel::getUuid()
					{
						return uuid;
					}

					void UserModel::setUuid(std::string uuid)
					{
						this->uuid = uuid;
					}

					std::string UserModel::getName()
					{
						return name;
					}

					void UserModel::setName(std::string name)
					{
						this->name = name;
					}

					int UserModel::getAge()
					{
						return age;
					}

					void UserModel::setAge(int age)
					{
						this->age = age;
					}

					std::string UserModel::ToString()
					{
						return "uuid="+uuid+",name="+name+",age="+age;
					}
				}
			}
		}
	}
}

File: 品味Java的21種設計模式/dp_java/templatemethod/example6/UserModel.java
package cn.javass.dp.templatemethod.example6;
/**
 * 描述用户的数据模型
 */
public class UserModel {
    private String uuid;
    private String name;
    private int age;
    public String getUuid() {
        return uuid;
    }
    public void setUuid(String uuid) {
        this.uuid = uuid;
    }
    public String getName() {
        return name;
    }
    public void setName(String name) {
        this.name = name;
    }
    public int getAge() {
        return age;
    }
    public void setAge(int age) {
        this.age = age;
    }
    public String toString() {
        return "uuid=" + uuid + ",name=" + name + ",age=" + age;
    }
}
